docs/USING_PUBLIC_CLOUDS.rst

```diff
@@ -0,0 +1,255 @@
+
+.. figure:: ../res/aws-console-versioning-enabled.png
+   :alt: AWS Console showing versioning enabled
+
+Microsoft Azure as a backend
+----------------------------
+
+locationConfig.json
+~~~~~~~~~~~~~~~~~~~
+
+Add a new region, which should describe the MS Azure hosted container you'll be
+writing your data to. There are a few configurable options here:
+
+- :code:`type` : set to :code:`azure` to indicate this region is hosted in MS Azure;
+- :code:`legacyAwsBehavior` : set to :code:`false`;
+- :code:`azureBlobEndpoint` : set to your storage account's endpoint, usually
+  :code:`https://{{storageAccountName}}.blob.core.windows.name`;
+- :code:`azureContainerName` : set to an *existing container* in your MS Azure
+  storage account; this is the container that will host all the data written in
+  this region;
+- :code:`bucketMatch` : set to :code:`true` if you want your object name to be
+  the same in your local bucket and your MS Azure container; set to
+  :code:`false` if you want your object name to be of the form
+  :code:`{{localBucketName}}/{{objectname}}` in your MS Azure container
+- :code:`azureBlobSAS` : provide a generated Shared Access Signature *without*
+  the initial question mark;
+
+.. code:: json
+
+    (...)
+    "azure-test": {
+	"type": "azure",
+        "legacyAwsBehavior": false,
+        "details": {
+          "azureBlobEndpoint": "https://zenkomeetups.blob.core.windows.net/",
+	  "bucketMatch": true,
+	  "azureBlobSAS": "sv=2017-04-17&ss=b&srt=sco&sp=rwdlac&se=2017-04-17T00:13:19Z&st=2017-10-10T16:13:19Z&spr=https&sig=R%2FUNb%2BjVrKrOsHyr%2F8V%2FpD%2F18%2BgUeYopsHxH%2Fjre9XNc%3K",
+	  "azureContainerName": "zenkontainer"
+	}
+    },
+    (...)
+
+Start the server with the ability to write to MS Azure
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Inside the repository, once all the files have been edited, you should be able
+to start the server and start testing pushing to MS Azure.
+
+.. code:: shell
+
+   # Start the server locally
+   $> S3DATA=multiple npm start
+
+Run the server as a docker container with the ability to write to MS Azure
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Mount all the files that have been edited to override defaults, and do a
+standard Docker run; then you can start testing pushing to MS Azure.
+
+.. code:: shell
+
+   # Start the server in a Docker container
+   $> sudo docker run -d --name CloudServer \
+   -v $(pwd)/data:/usr/src/app/localData \
+   -v $(pwd)/metadata:/usr/src/app/localMetadata \
+   -v $(pwd)/locationConfig.json:/usr/src/app/locationConfig.json \
+   -v $(pwd)/conf/authdata.json:/usr/src/app/conf/authdata.json \
+   -e S3DATA=multiple -e ENDPOINT=http://localhost -p 8000:8000
+   -d scality/s3server
+
+Testing: put an object to MS Azure using CloudServer
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In order to start testing pushing to MS Azure, you will need to create a local
```

Comment:
Again, same as with AWS, you don't need to create a bucket in the Azure location for the objects to be put there. It can be done at the object level